src/main/java/seedu/duke/parsers/AddParser.java

```diff
@@ -3,28 +3,58 @@
 import seedu.duke.commands.AddCommand;
 import seedu.duke.commands.Command;
 import seedu.duke.exceptions.ModHappyException;
+import seedu.duke.exceptions.ParseException;
 
+import java.util.HashMap;
+
+/**
+ * This Parser supports the "add" command.
+ */
 public class AddParser extends Parser {
+    private static final String FLAG = "flag";
+    private static final String TASK_NAME = "taskName";
+    private static final String TASK_DESCRIPTION = "taskDescription";
+    private static final String MODULE_CODE = "moduleCode";
+    private static final String MODULE_DESCRIPTION = "moduleDescription";
 
-    private static final String ADD_COMMAND_WORD = "add";
-    private static final String FLAG = "\t";
-    private static final String COMMAND_WORD = "commandWord";
-    private static final String ADD_FORMAT = "\\s*(?<flag>(\\/(m|t)))\\s"
-            + "+(?<argument1>[^\\-]*)\\s*((?<subFlag>\\-(\\bmod|d\\b))\\s+(?<argument2>.+))*";
+    // Unescaped regex for testing (split into two lines):
+    // ^\s*(\/t\s+(?<taskName>.+?(?=(\s+-d\s+)|$))(\s+(-d\s+(?<taskDescription>.+)))?|
+    // \/m\s+(?<moduleCode>\w+?(?=(\s+-d\s+)|$))(\s+(-d\s+(?<moduleDescription>.+)))?)
+    // TODO: Add support for -mod argument when integrating Task and Module classes with one another
+    private static final String ADD_FORMAT = "^\\s*(\\/t\\s+(?<taskName>.+?(?=(\\s+-d\\s+)|$))"
+            + "(\\s+(-d\\s+(?<taskDescription>.+)))?|\\/m\\s+(?<moduleCode>\\w+?(?=(\\s+-d\\s+)|$))"
+            + "(\\s+(-d\\s+(?<moduleDescription>.+)))?)";
```

Comment:
The original format already support the mod. May I check why it is changed?

Reply:
I changed it to fix #71 and also to future-proof the regex in case `add /t` and `add /m` end up having significantly different syntax.